src/main/java/io/cryostat/templates/TemplatesModule.java

```diff
@@ -55,4 +56,17 @@ static LocalStorageTemplateService provideLocalStorageTemplateService(
             FileSystem fs, Environment env) {
         return new LocalStorageTemplateService(fs, env);
     }
+
+    @Provides
+    @Singleton
+    static LocalProbeTemplateService provideLocalProbeTemplateService(
+            FileSystem fs, Environment env) {
+        try {
+            return new LocalProbeTemplateService(fs, env);
+        } catch (Exception e) {
+            // Dagger doesn't like constructors that can throw exceptions, the probeTemplateService
+            // throws an exception if the sanity checks fail so we need to deal with it here
+            return null;
```

Comment:
Dagger doesn't like when `@Provides` methods throw checked exceptions, but it'll let you throw an unchecked exception. This could be:

```java
try {
  return new LocalProbeTemplateService(fs, env);
} catch (Exception e) {
  throw new RuntimeException(e);
}
```

and it would compile and work as expected. This pattern is used in a few places already, for example:

https://github.com/cryostatio/cryostat/blob/059e6de8ac450a162c130285dc34c3375ec6dd66/src/main/java/io/cryostat/net/web/WebModule.java#L79

This way, wherever you are trying to inject the `LocalProbeTemplateService` doesn't have to do a null check. If we're at the point of instantiating the dependent class then the sanity checks must have already passed, and the dependency is non-null.

Reply:
Makes sense, I've cleaned this up, thanks